sunpy/net/helioviewer.py

```diff
@@ -207,19 +255,19 @@ def download_png(self, date, image_scale, layers, directory=None,
         >>> from sunpy.net.helioviewer import HelioviewerClient
         >>> import datetime
         >>> hv = HelioviewerClient()  # doctest: +REMOTE_DATA
-        >>> file = hv.download_png('2012/07/16 10:08:00', 2.4, "[SDO,AIA,AIA,171,1,100]", x0=0, y0=0, width=1024, height=1024)   # doctest: +REMOTE_DATA
-        >>> file = hv.download_png('2012/07/16 10:08:00', 4.8, "[SDO,AIA,AIA,171,1,100],[SOHO,LASCO,C2,white-light,1,100]", x1=-2800, x2=2800, y1=-2800, y2=2800)   # doctest: +REMOTE_DATA
-        >>> file = hv.download_jp2(datetime.datetime.now(), observatory='SDO', instrument='HMI', detector='HMI', measurement='continuum')   # doctest: +REMOTE_DATA
+        >>> file = hv.download_png('2012/07/16 10:08:00', 2.4, "[SDO,AIA,AIA,171,1,100]", False, x0=0, y0=0, width=1024, height=1024)   # doctest: +REMOTE_DATA
+        >>> file = hv.download_png('2012/07/16 10:08:00', 4.8, "[SDO,AIA,AIA,171,1,100],[SOHO,LASCO,C2,white-light,1,100]", True, x1=-2800, x2=2800, y1=-2800, y2=2800)   # doctest: +REMOTE_DATA
         """
         params = {
             "action": "takeScreenshot",
             "date": self._format_date(date),
             "imageScale": image_scale,
             "layers": layers,
+            "eventLabels": eventLabels,
             "display": True
         }
-        params.update(kwargs)
 
+        params.update(kwargs)
```

Comment:
A problem I see by having the optional parameters as `kwargs` is that you won't know whether you get them wrong nor your IDE would be able to help you to add them.

Reply:
So do we need to add a test to check if they entered the correct option

Reply:
We could. You want to fail early and hard. Say for example the `directory` entry. If someone calls it with a number, then it will fail once it has downloaded the files and try to save it in the computer. Imagine that file is 1 GB and you have been waiting 20 minutes to download. At the end you've been angry of why the program didn't tell you before starting the download. (Note, this is an example, I've not checked whether if you put a number it fails or it creates a directory with such name).

Reply:
Ok, this might take some time. I will incorporate rest of the changes

Reply:
@Cadair @nabobalis - what do you think about the `**kwargs` in this case?

Reply:
I was against it at first since I thought if you want to use any keywords we should support them explicitly. But there are parts of the API we never want to actually support. But might be useful to someone who had more knowledge of the API and we can support that via kwargs

Reply:
Then we should decide which one we support, give explicitly the default values for these and left the others open for the "experts".

Reply:
Oh, so which all do we support explicitly?

Reply:
We will have to decide that.

Reply:
Ok so what all according to you should we support?

Reply:
I don't know. Its a big question.I would say that we should support the most basic keywords that allow the function to work. But beyond that, we would need to discuss it.

Reply:
Yeah, it is a big decision. Let's see what others have to say about it

Reply:
I have done all the changes except the one concerned with the optional parameters.
I am not able to push my commit. Force push will disturb the commit history

Reply:
I am unsure what you mean? You already went back to the older history. So you shouldn't be force pushing over?

Reply:
Actually I was trying to push my commits which incorporates all the changes except that one but when I push it on git it says rejected

Reply:
If you have changed the history, you have to force push. I'm not aware of any other way.